src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -550,10 +556,6 @@ async def _setup_constraints_repository_pex(
     global_requirement_constraints: GlobalRequirementConstraints,
 ) -> OptionalPexRequest:
```

Comment:
If my read is correct, there's no reason for this to be optional anymore

Reply:
It can still bail below `if unconstrained_projects`.